### PR TITLE
Add BXML Builder

### PR DESCRIPTION
**Pull request by illiriks**
Added the BXML builder, along with tests for it.

### CODE REVIEW

**Comment by dtolb** [comment]
@nguyer @a-type THOUGHTS!!!???

**Comment by a-type** [comment]
I find it amusing that you wrote out functional code _and_ expected output, but not real unit tests... so close... :confused: 

I find the control flow to be a bit confusing after an initial overview, but I think that's more due to BXML's design than the tool. In fact, compared to a group of nested nodes, the `actionAnd` pattern makes it a bit more obvious that the internal actions will occur after the containing one has completed.

I'm normally not the biggest fan of using callbacks in this day and age. But this seems to be a valid use case. I can't think of any obviously better approach, certainly.

Would be even cooler if we could transpile JSX with a custom Babel transform so you could drop the XML right in the code... that's straight overboard though. Maybe a sabbatical idea :laughing:

**Comment by illiriks** [comment]
Added unit tests @a-type. I also renamed `gatherAnd` et al. to `gather` as using simply the XML verb seems to be best practice.

**Comment by a-type** [comment]
The constants in your unit test file get kinda messy. I'd recommend just storing them in fixture files (`test/fixtures/basicCall.xml`) and reading them in from the file system.

**Comment by a-type** [comment]
Cool, lgtm.

File: test/bxml-responses/hangup.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Response>
+     <Hangup/>
```
Comment (dtolb) [inline test/bxml-responses/hangup.xml:3]: does this need to be [`<hangup></hangup>`](http://ap.bandwidth.com/docs/xml/hangup/) ?

**Comment by dtolb** [comment]
Can you add some examples to the jsdoc?... Looks like you can almost just use the tests. I wonder if there is a way to link the test code as the example :question:

File: lib/xml.js [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+	 * @param {function} [callback] A function containing the verbs to be nested inside the Call verb
+	 * @return {BXMLResponse} this, for chaining
+	 * @example
+	 * //This app will create a call and tell the callee they are being called.
+	 * var myApp = new BXMLResponse();
+	 * myApp.call({
+	 *      from : '+19195551212',
+	 *      to   : '+19195551213'
+	 * }, function () {
+	 *      this.speakSentence("You are recieving a call from 919 555 1212.');
```
Comment (dtolb) [inline lib/xml.js]: oh snap! missed that `"` with a `'`

File: test/bxml-responses/hangup.xml (hunk 1 above)
Comment (illiriks) [inline test/bxml-responses/hangup.xml:3]: That's a standard empty-element tag in XML - should work. I can make it `<hangup></hangup>` though

File: lib/xml.js (hunk 2 above)
Comment (illiriks) [inline lib/xml.js]: fixed

**Pull request merged**